ape_hardhat/exceptions.py

```diff
@@ -81,24 +102,35 @@ def check(self):
             self._is_running = False
 
             if isinstance(self._exception, type):
-                raise self._exception(str(self))
+                raise self._error(self._exception(str(self)))
 
             elif isinstance(self._exception, Exception):
-                raise self._exception
+                raise self._error(self._exception)
 
             raise self
 
     def cancel(self):
         self._is_running = False
 
+    def _error(self, err: Exception) -> Exception:
+        self._process.stop()
```

Comment:
fixes issue where process was not stopping when failing early

Reply:
Why not turn this into a context manager?

Reply:
It is! https://github.com/ApeWorX/ape-hardhat/blob/main/ape_hardhat/process.py#L159

Reply:
This kind of process cleanup would be something I'd expect in a ContextManager's `__enter__`/`__exit__` methods

Reply:
Ah, I see what you mean.

Reply:
for reference: https://docs.python.org/3/library/stdtypes.html#typecontextmanager

Reply:
This class _is_ a context manager, the poll closing part just was not happening in the `__exit__`.
But I made it so now.